packages/dds/sequence/src/sharedString.ts

```diff
@@ -260,4 +280,14 @@ export class SharedString extends SharedSegmentSequence<SharedStringSegment> imp
     public getMarkerFromId(id: string): ISegment {
         return this.client.getMarkerFromId(id);
     }
+
+    protected rollback(content: any, localOpMetadata: unknown): void {
+        if (this.previousInsertsCount > 0 && content === this.previousInserts[this.previousInsertsCount - 1]) {
+            const msg = content as IMergeTreeInsertMsg;
+            this.removeRange(msg.pos1, msg.pos1 + (msg.seg as string).length, false);
```

Comment:
this may cause problems. i don't think you can just call this.removeRange. you probably need to call a new method on this.client. The method will need to call into merge tree and dequeue the pending segment group. see ackPendingSegment which is called on incoming op